ql/patterns/observable.hpp

```diff
@@ -110,7 +111,7 @@ namespace QuantLib {
         virtual ~Observer();
 
         // observer interface
-        std::pair<iterator, bool>
+        std::pair<boost::unordered_set<ext::shared_ptr<Observable>>::iterator, bool>
```

Comment:
Deprecating `set_type` doesn't help if we have `boost::` exposed in the interface explicitly.

I would deprecate `set_type` (and make it private later) but leave the `iterator` typedef alone (and public) so it can be used in this signature.

Reply:
Done